tests/js/fixtures/detectors.ts

```diff
@@ -0,0 +1,35 @@
+import {DataConditionGroupFixture} from 'sentry-fixture/dataConditions';
+import {UserFixture} from 'sentry-fixture/user';
+
+import type {Detector} from 'sentry/types/workflowEngine/detectors';
+
+export function DetectorFixture(params: Partial<Detector>): Detector {
+  return {
+    id: '1',
+    name: 'detector',
+    projectId: '1',
+    createdBy: UserFixture().id,
+    dateCreated: new Date('2025-01-01T00:00:00.000Z'),
+    dateUpdated: new Date('2025-01-01T00:00:00.000Z'),
+    lastTriggered: new Date('2025-01-01T00:00:00.000Z'),
+    workflowIds: [],
+    config: {},
+    type: 'metric',
+    dataCondition: DataConditionGroupFixture({}),
+    disabled: false,
+    dataSource: {
+      id: '1',
+      status: 1,
+      snubaQuery: {
+        aggregate: '',
+        dataset: '',
+        id: '',
+        query: '',
+        timeWindow: 60,
+        ...params.dataSource?.snubaQuery,
+      },
+      ...params.dataSource,
```

Comment:
because the type is Partial<Detector> people won't likely pass just part of this object and will replace the entire thing. Could also pull data source into another fixture in this file or something
```suggestion
      },
```

Reply:
Added a DataSource fixture here 8a5b4f4b0cec49b6b75b12dfa417ad35520e3c95